### DISCUSSION TITLE
Error while trying to import pysyft in lesson 1 Duet Basics (Foundations course)

### DISCUSSION THREAD

**Discussion by cbs-rschreijen** [OpenMined/courses#273] [Course: Foundations of Private Computation] [answered]
Hi, 

I setup a virtual environment (tried both conda and venv), activated it and installed pip requirements per course instruction.  When running the notebooks for the Duet Basics demo I get an error at the first statement in both notebooks while loading the pysyft library (same error). I am running on macOS 11.1 Big Sur. 

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-281ee5c8ca5f> in <module>
----> 1 import syft as sy

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/__init__.py in <module>
     59 from syft.core.common.serde.serialize import _serialize as serialize  # noqa: F401
     60 from syft.core.node.common.service.repr_service import ReprMessage  # noqa: F401
---> 61 from syft.core.node.device.device import Device  # noqa: F401
     62 from syft.core.node.device.device import DeviceClient  # noqa: F401
     63 from syft.core.node.domain.domain import Domain  # noqa: F401

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/core/node/device/__init__.py in <module>
      1 # syft relative
----> 2 from .client import DeviceClient
      3 from .device import Device
      4 
      5 __all__ = ["DeviceClient", "Device"]

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/core/node/device/client.py in <module>
     14 from ...io.location import SpecificLocation
     15 from ...io.route import Route
---> 16 from ..common.client import Client
     17 
     18 

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/core/node/common/client.py in <module>
     18 from ....core.pointer.pointer import Pointer
     19 from ....decorators import syft_decorator
---> 20 from ....lib import lib_ast
     21 from ....proto.core.node.common.client_pb2 import Client as Client_PB
     22 from ....proto.core.node.common.metadata_pb2 import Metadata as Metadata_PB

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/lib/__init__.py in <module>
     24 
     25 # constructor: copyType = create_lib_ast
---> 26 lib_ast = create_lib_ast()
     27 lib_ast._copy = create_lib_ast

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/lib/__init__.py in create_lib_ast()
     10 
     11     python_ast = create_python_ast()
---> 12     torch_ast = create_torch_ast()
     13     torchvision_ast = create_torchvision_ast()
     14     # numpy_ast = create_numpy_ast()

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/lib/torch/__init__.py in create_torch_ast()
     51                 # this allows us to import them for testing
     52                 continue
---> 53             ast.add_path(
     54                 path=method, framework_reference=torch, return_type_name=return_type
     55             )

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/ast/globals.py in add_path(self, path, index, return_type_name, framework_reference)
     63         attr = self.attrs[framework_name]
     64         if hasattr(attr, "add_path"):
---> 65             attr.add_path(  # type: ignore
     66                 path=path, index=1, return_type_name=return_type_name
     67             )

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/ast/module.py in add_path(self, path, index, return_type_name, framework_reference)
    118             self.lookup_cache[attr_ref] = path
    119         if hasattr(attr, "add_path"):
--> 120             attr.add_path(  # type: ignore
    121                 path=path, index=index + 1, return_type_name=return_type_name
    122             )

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/ast/callable.py in add_path(self, path, index, return_type_name)
     80             if path[index] not in self.attrs:
     81 
---> 82                 attr_ref = getattr(self.ref, path[index])
     83 
     84                 if isinstance(attr_ref, module_type):

AttributeError: type object 'Tensor' has no attribute 'fft'

**Comment by mcleonard** [comment, marked as answer]
Hi there! Turns out the version of Syft on PyPI is a bit old. Can you install Syft with this:

`pip install git+https://github.com/OpenMined/PySyft@dev#egg=syft`

You might need to uninstall Syft first, `pip uninstall syft`.

This should work, let me know if it doesn't.

**Reply by cbs-rschreijen** [reply]
Thank you mcleonard for the quick reply. 

Your fix worked. `Pip uninstall syft` removed version 0.3.0 from my system. Your pip install command reinstalled version syft-0.3.0.post0.dev754+g65a4aa120. Restarting the notebooks and rerunning the `import syft as sy` now runs without errors.

I suggest fixing the requirements.txt file or changing the instructions in the course as others will probably run into the same issue.

Thanks again, Ralph

**Reply by mcleonard** [reply]
Yep, already fixed!

**Reply by RushnaiwalaTaki** [reply]
Yess. Its fixes. Great, Thanks

**Reply by pengdan1129** [reply]
It doesn't work to me! please help

**Reply by Merna-Atef** [reply]
It doesn't work for me either. I have tried all the steps.

**Reply by YasMinSdt** [reply]
> It doesn't work to me! please help

Have you found a solution to this?

**Reply by sudharsan-g** [reply]
@YasMinSdt
Try installing  jaxlib==0.3.14 first if your using it in windows 
![Screenshot 2023-04-04 131457](https://user-images.githubusercontent.com/66720023/229723690-bd42619b-6c59-494a-9b44-138797b3504e.png)
I'm also in the same point where u were,new to pysyft.Hope this would help
